server/sonar-server/src/main/java/org/sonar/server/rule/index/RuleIndex.java

```diff
@@ -494,8 +524,32 @@ private static void setPagination(SearchOptions options, SearchRequestBuilder es
     return terms;
   }
 
+  public Set<String> listTags(OrganizationDto organization, @Nullable String query, int size) {
+    TermsQueryBuilder scopeFilter = QueryBuilders.termsQuery(
+      FIELD_RULE_EXTENSION_SCOPE,
+      RuleExtensionScope.system().getScope(),
+      RuleExtensionScope.organization(organization).getScope());
+
+    TermsBuilder termsAggregation = AggregationBuilders.terms(AGGREGATION_NAME_FOR_TAGS)
+      .field(FIELD_RULE_EXTENSION_TAGS)
+      .size(size)
```

Comment:
I added order. This will change the behavior of api/rules/tags - but I think it is an improvement for every usecase.